### PR TITLE
Hypervalidation

### LINKED ISSUE #9: Add one cycle LR to hypervalidation
- [x] Add OneCycleLR https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.OneCycleLR.html
- [x] Calculate learning rate using LR finder: https://pytorch-lightning.readthedocs.io/en/1.4.5/advanced/lr_finder.html
- [x] Add "epoch_num" hyperparameter to trial and remove early stopping
- [x] Refactor LR to separate CLI
- [x] Run LR finder in Trial and add it to model train runs

### PR DESCRIPTION
**Pull request by sternakt**
Add hypervalidation command to temporal data kit

Closes #9 #10 #12

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ repos:
         name: Linter
         entry: "scripts/lint-pre-commit.sh"
         language: python
-        language_version: python3.8
+        language_version: python3.10
```
Comment (davorrunje) [inline .pre-commit-config.yaml:25]: why this? lower versions are needed to make sure it is compatible with everything

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -33,7 +33,7 @@ repos:
         name: Static analysis
         entry: "scripts/static-pre-commit.sh"
         language: python
-        language_version: python3.8
+        language_version: python3.10
```
Comment (davorrunje) [inline .pre-commit-config.yaml:36]: same thing, but for mypy

File: temporal_data_kit/cli.py [hunk 3]
```diff
@@ -0,0 +1,63 @@
+__all__ = ["app", "run_study", "hello"]
```
Comment (davorrunje) [inline temporal_data_kit/cli.py:1]: should be sorted, linter should do that

File: .pre-commit-config.yaml (hunk 1 above)
Comment (sternakt) [reply .pre-commit-config.yaml]: Reverted version to 3.8, works fine, I guess something got corrupted on my side temporarily in pre-commit cache

File: .pre-commit-config.yaml (hunk 2 above)
Comment (sternakt) [reply .pre-commit-config.yaml]: Same as above

File: temporal_data_kit/cli.py (hunk 3 above)
Comment (sternakt) [reply temporal_data_kit/cli.py]: Ran linter in pre-commit, it didn't sort it

**Review by davorrunje — APPROVED** [review]
